internal/db/modelmigration.go

```diff
@@ -24,14 +27,60 @@ func (d *Database) AddIncomingModelMigration(ctx context.Context, modelMigration
 	defer durationObserver()
 	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
 
+	lookup := dbmodel.IncomingModelMigration{
+		ModelUUID: modelMigration.ModelUUID,
+	}
+	// Set noWait to true to ensure that if the row is locked by another transaction,
+	// we will return an error immediately instead of waiting.
+	err = d.GetIncomingModelMigrationWithLock(ctx, &lookup, true)
+	if err == nil {
+		// If the model migration already exists, update it.
+		modelMigration.ID = lookup.ID
+	} else if err != nil && errors.ErrorCode(err) != errors.CodeNotFound {
+		return errors.E(op, err)
+	}
+
 	db := d.DB.WithContext(ctx)
 
-	if err := db.Create(modelMigration).Error; err != nil {
+	if err := db.Save(modelMigration).Error; err != nil {
```

Comment:
IIRC using db.Model(x).Save(...) can lead to no updated_at field beign set, I'm guessing this is fine?